phoenix-core/src/main/java/org/apache/phoenix/mapreduce/index/PhoenixIndexImportDirectReducer.java

```diff
@@ -49,8 +50,42 @@
     private static final Logger LOGGER =
             LoggerFactory.getLogger(PhoenixIndexImportDirectReducer.class);
 
+    private static final String SPLIT_MERGE_ERROR_STR_FMT =
+            "Data table split/merge happened during Index rebuild. Old region cnt=%d, New region cnt = %d";
+
+    private static Integer regionCountForTesting = null;
+
+    public static void setRegionCountForTesting(Integer regionCount) {
+        regionCountForTesting = regionCount;
+    }
+
     @Override
     protected void cleanup(Context context) throws IOException, InterruptedException{
+        String strNumOfRegions = PhoenixConfigurationUtil.getNumOfRegions(context.getConfiguration());
```

Comment:
Isn't this the number of mappers? Can the reducer get it instead and compare with it?

Reply:
It is not mappers. It is table region count. Not sure I understand what you are suggesting